src/datasources/cache/cache.first.data.source.spec.ts

```diff
@@ -149,4 +149,48 @@ describe('CacheFirstDataSource', () => {
 
     expect(mockCache.set).toHaveBeenCalledWith(cacheDir, expect.anything(), 30);
   });
+
+  it('should cache not found errors with a specific TTL', async () => {
+    const mockCache = jest.mocked({
+      get: jest.fn(),
+      set: jest.fn(),
+      delete: jest.fn(),
+    } as unknown as ICacheService);
+
+    cacheFirstDataSource = new CacheFirstDataSource(
+      mockCache,
+      mockNetworkService,
+      mockLoggingService,
+    );
```

Comment:
The test setup here looks a bit strange. On one hand we have the `beforeEach` which already sets up a `CacheFirstDataSource`. But we ignore it here to setup a different one.

Reply:
I think this is wired that way to allow using a spy as `CacheService` and make assertions on the calls made. I copied the setup from [the test above this one](https://github.com/safe-global/safe-client-gateway-nest/blob/06dfc1a6e2b13ed39abb9a2b18f3b9fe30d4afdc/src/datasources/cache/cache.first.data.source.spec.ts#L120). Maybe we could think of a better way to achieve this, but I'd say it could be in a different PR, wdyt?

Reply:
Ok let's consider rewriting this in a separate PR 👍